custom-catalog/catalog_builder.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/python3
+
+import os
+import json
+import toml
+import shutil
+import time
+import subprocess
+from collections import OrderedDict
+
+
+now = time.time()
+my_env = os.environ.copy()
+my_env["GIT_TERMINAL_PROMPT"] = "0"
+
+DEFAULT_APPS_FOLDER = "/ynh-dev/custom-apps/"
+DEFAULT_APP_BRANCH = "master"
+
+
+def app_input_folder(folder, app):
+    return os.path.join(folder, app + "_ynh")
+
+
+def app_cache_folder(folder, app):
+    return os.path.join(folder, ".apps_cache", app)
+
+
+def clone_app(app, branch, path, dest):
+    cmd = f"git clone --quiet --depth 1 --single-branch --branch {branch} file://{path} {dest}"
+    return subprocess.check_output(cmd.split(), env=my_env).strip().decode("utf-8")
+
+
+def refresh_all_caches(app_list, folder):
+    cache_folder = os.path.join(folder, ".apps_cache")
+    os.makedirs(cache_folder, exist_ok=True)
+
+    for app, infos in app_list.items():
+        app = app.lower()
+        branch = infos.get("branch", DEFAULT_APP_BRANCH)
+        path = infos.get("url") or app_input_folder(folder, app)
+        dest = app_cache_folder(folder, app)
+
+        if os.path.exists(dest):
+            shutil.rmtree(dest)
+
+        print(f"Cloning {app}[{branch}] from '{path}'...")
+
+        try:
+            clone_app(app, branch, path, dest)
+        except Exception as e:
+            raise f"Failed to clone '{app}' {str(e)}"
```

Comment:
I guess that works but I wonder if we couldnt remove all this entirely to simplify the code? Maybe that would also prevent the necessity from comitting changes ?

Reply:
Yep ! Was probably overthinking dXD. But you'd still need to commit changes even if we use directly the `custom-apps/app` since `_extract_app()` will clone from a branch

Reply:
Removed the useless part ; )